### PR TITLE
DOCSP-13806: add note about $uuid parsing in extended json

### PR DESCRIPTION
**Pull request by biniona-mongodb**
## Pull Request Info
Adds note about `$uuid` extended json type parsing.

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13806

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/a0e24f5/java/docsworker-xlarge/DOCSP-13806/fundamentals/data-formats/document-data-format-extended-json/#extended-json-formats

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/fundamentals/data-formats/document-data-format-extended-json.txt [hunk 1]
```diff
@@ -70,6 +70,16 @@ See the table below to see a description of each format:
...
 .. _extended_json_example_section:
 
+.. note::
+
+   The ``$uuid`` Extended JSON type is parsed from a string to a
+   :java-docs:`BsonBinary </apidocs/bson/org/bson/BsonBinary.html#<init>(java.util.UUID)>`
+   object of binary subtype 4. For a further discussion of ``$uuid`` field
+   parsing, see the
+   :spec:`special rules for parsing $uuid fields </extended-json.rst#special-rules-for-parsing-uuid-fields>`
+   section in the extended JSON specification.
```
Comment (terakilobyte) [inline source/fundamentals/data-formats/document-data-format-extended-json.txt:80]:
```suggestion
   The ``$uuid`` Extended JSON type is parsed from a String to a
   :java-docs:`BsonBinary </apidocs/bson/org/bson/BsonBinary.html#<init>(java.util.UUID)>`
   object of binary subtype 4. For more information about ``$uuid`` field
   parsing, see the
   :spec:`special rules for parsing $uuid fields </extended-json.rst#special-rules-for-parsing-uuid-fields>`
   section in the extended JSON specification.
```

**Review by terakilobyte — CHANGES_REQUESTED** [review]
I think "more information about" is more simple than "a further discussion of", and we should capitalize types if they are capitalized in the language.

File: source/fundamentals/data-formats/document-data-format-extended-json.txt (hunk 1 above)
Comment (biniona-mongodb) [reply source/fundamentals/data-formats/document-data-format-extended-json.txt:80]: Thanks for the notes. I agree with both (in addition "for more information..." is used often elsewhere in the docs). Committing your suggestion. 

Going forward, I will try to stress simple sentences (and parallel structure), and appropriately capitalize language types.

**Review by terakilobyte — APPROVED** [review]
LGTM. Is this going for external review?

**Comment by biniona-mongodb** [comment]
> LGTM. Is this going for external review?

I think Jeff should probably also sign off on it just to make sure I got all my terms correct. I don't think it is specified anywhere if an external review is required though.

**Comment by biniona-mongodb** [comment]
Hi @jyemin, can I please have a technical review of this PR?

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**